Add docs for nested claims in JWT RBAC (#5190)
solo-io/gloo: docs/content/guides/security/auth/jwt/access_control.md

- the request URI starts with `/api/pets`
- the request contains a verifiable JWT
- the JWT has a `sub` claim with value `system:serviceaccount:default:svc-a`

>   - **Note**: It is also possible to match against nested claims (claims that are not at the top-level of the JWT, but instead are a child of another claim).
>     This functionality is disabled by default, but can be enabled by specifying a `nestedClaimsDelimiter` (i.e. `.`) on a [Policy]({{% versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/enterprise/options/rbac/rbac.proto.sk/#policy" %}})
>     and specifying the claim name as a path (i.e. `parent.child.foo`) in the [JWTPrincipal]({{% versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/enterprise/options/rbac/rbac.proto.sk/#jwtprincipal" %}}).

Reviewer: ```suggestion
  - **Note**: Matching is supported for top-level claims of the JWT and for nested claims, or claims that are children of top-level claims.
    To enable matching against nested claims, you must first specify a `nestedClaimsDelimiter`, such as `.`, in a [Policy]({{% versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/enterprise/options/rbac/rbac.proto.sk/#policy" %}})
    and specify the claim name as a path, such as `parent.child.foo`, in the `claims` field of the [JWTPrincipal]({{% versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/enterprise/options/rbac/rbac.proto.sk/#jwtprincipal" %}}).
```
